packages/opencensus-core/src/trace/instrumentation/base-plugin.ts

```diff
@@ -60,10 +78,59 @@ export abstract class BasePlugin implements types.Plugin {
   // while unpatching is not. It should be the other way around
 
   // tslint:disable:no-any
-  applyPatch(moduleExports: any, tracer: modelTypes.Tracer, version: string):
-      any {
-    this.setPluginContext(moduleExports, tracer, version);
+  applyPatch(
+      moduleExports: any, tracer: modelTypes.Tracer, version: string,
+      basedir?: string): any {
+    this.setPluginContext(moduleExports, tracer, version, basedir);
   }
 
   abstract applyUnpatch(): void;
+
+
+  /**
+   * Load internal files according to version range
+   */
+  private loadInternalFiles(): types.ModuleExports {
+    let result: types.ModuleExports = null;
+    this.logger.debug('loadInternalFiles %o', this.internalFileList);
+    if (this.internalFileList) {
+      Object.keys(this.internalFileList).map(versionRange => {
+        if (semver.satisfies(this.version, versionRange)) {
+          result = this.loadInternalModuleFiles(
+              this.internalFileList[versionRange], this.basedir);
+        }
+        if (!result) {
+          this.logger.debug(
+              'No internal file could be loaded for %s@%s', this.moduleName,
+              this.version);
+        }
+      });
+    }
+    return result;
+  }
+
+
+  /**
+   * Load internal files from a module and  set internalFilesExports
+   */
+  private loadInternalModuleFiles(
+      extraModulesList: types.PluginNames,
+      basedir: string): types.ModuleExports {
+    const extraModules: types.ModuleExports = {};
+    if (extraModulesList) {
+      Object.keys(extraModulesList).map(modulename => {
+        try {
+          this.logger.debug('loading File %s', extraModulesList[modulename]);
+          extraModules[modulename] =
+              require(path.join(basedir, extraModulesList[modulename]));
```

Comment:
I am not certain that this will work in all cases. What does `basedir` usually look like -- is it absolute or relative?

Reply:
The basedir parameter is passed to the callback of the hook function from lib "require-in-the-middle", it is an absolute path.